### ISSUE TITLE
[Feature] Add validation for name field against aliases

### ISSUE THREAD

**Issue by DogmaDragon** [stashapp/stash-box#726]
**Is your feature request related to a problem? Please describe.**
Right now, stash-box allows adding an identical alias as the name. Since search and scraping already matches both, I feel like this is redundant clutter. 

**Describe the solution you'd like**
Add validation for name field against aliases in the frontend.

**Additional context**
![image](https://github.com/stashapp/stash-box/assets/103123951/3f97318a-0fe7-43db-bbb5-32ef4aae97a6)

**Comment by AdultSun** [comment]
Also worth noting that Stash already prevents identical aliases. I've seen others mention this causing errors when trying to scrape from a Stash-Box into Stash, but I haven't run into it personally yet.

**Comment by DogmaDragon** [comment]
True, but only if they have the same capitalization. That's a different issue, https://github.com/stashapp/stash/issues/4250.

**Comment by echo6ix** [comment]
I had a different take on resolving this issue earlier this year https://github.com/stashapp/stash-box/issues/610

TLDR:
- Consolidate the `name` and `alias` fields into one *Names* field
- Present the names using [filter chips](https://m3.material.io/components/chips/specs#a93c9762-7897-427c-bc49-94942a4a0767) styling
  - Filter chips look exactly the same as the incumbent chips, but they add a trailing drop down caret icon
  - This gives each chip the option to `Set as primary name` and `Delete name`
  - A name set as primary gets a unique chip presentation to stand out
     - Either a different colored chip
     - Or a chip with a leading icon, such as a star

![image](https://github.com/stashapp/stash-box/assets/37937507/c7f4b265-8865-4579-bb70-18d28b99a7b5)

This solves several issues with manual data entry such as the need for validation against duplicate aliases, and forgetting to re-add a primary name as an alias if the submitter is flipping primary names.

#### Alternate implementation
Thought of alternate implementation of the above without using a filter chip

![image](https://github.com/stashapp/stash-box/assets/37937507/3167da1a-c80a-4986-a65b-e93a66e44e32)
